internal/pkg/secret/encryption_other.go

```diff
@@ -0,0 +1,86 @@
+//go:build linux || darwin
+
+package secret
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"os"
+	"strconv"
+
+	"github.com/denisbrodbeck/machineid"
+
+	"golang.org/x/crypto/pbkdf2"
+)
+
+func GenerateRandomBytes(n int) ([]byte, error) {
+	b := make([]byte, n)
+	_, err := rand.Read(b)
+	if err != nil {
+		return []byte{}, err
+	}
+	return b, err
+}
+
+func DeriveEncryptionKey(salt []byte) ([]byte, error) {
+	machineId, err := machineid.ID()
+	if err != nil {
+		return []byte{}, err
+	}
+	userId := strconv.Itoa(os.Getuid())
+	encryptionKey := pbkdf2.Key([]byte(machineId+userId), salt, 10000, 32, sha256.New)
```

Comment:
What do the numbers 10000 and 32 represent?

Reply:
10000 is iteration, 32 is the key length